### PR TITLE
KAFKA-17626: Move common fetch related classes from storage to server-common

### PR DESCRIPTION
**Pull request by apoorvmittal10**
Moved classes which are common in fetch and exposed to share fetch in storage-api module.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: checkstyle/import-control-share.xml [hunk 1]
```diff
@@ -45,7 +45,7 @@
       <allow pkg="org.apache.kafka.server.share" />
 
       <subpackage name="fetch">
-        <allow class="org.apache.kafka.storage.internals.log.FetchParams"/>
```
Comment (mumrah) [inline checkstyle/import-control-share.xml:48]: Can you check other import control files to see if any others need cleanup?

**Review by mumrah — APPROVED** [review]
Thanks @apoorvmittal10, LGTM!

**Comment by chia7712** [comment]
Apologies, but I assume the code in storage-api is intended to serve as a plugin interface. For instance, the current storage-api code provides a pluggable API for remote storage. Similarly, tool-api offers a plugin interface for console tools, and group-coordinator-api is designed for server partition assignors.

Do the common fetch-related classes serve a similar purpose? Or is the main goal simply to move them into a "common" module? If that's the case, perhaps we could relocate them to server-common?

**Comment by mumrah** [comment]
@chia7712 I wasn't aware that storage-api was meant as a public API. I thought this module was just decoupling internal dependencies. Thanks for the correction 😅 (this change was my suggestion in a previous PR).

@apoorvmittal10 Seems like server-common is a better place after all.

File: checkstyle/import-control-share.xml (hunk 1 above)
Comment (apoorvmittal10) [reply checkstyle/import-control-share.xml:48]: I checked elsewhere but can't find anything to clean.

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by apoorvmittal10** [comment]
@mumrah @chia7712 Thanks for the review. The build is green.

**Comment by mumrah** [comment]
@apoorvmittal10 can you update the PR description?

**Pull request merged**